ibis/pyspark/tests/test_basic.py

```diff
@@ -163,6 +163,25 @@ def test_filter(client, filter_fn, expected_fn):
     tm.assert_frame_equal(result.toPandas(), expected.toPandas())
 
 
+@pytest.mark.parametrize('distinct', [False, True])
+def test_union(client, distinct):
+    table = client.table('basic_table')
+
+    result = table.union(table, distinct=distinct).compile().toPandas()
+
+    if distinct:
+        expected = pd.DataFrame({'id': range(0, 10), 'str_col': 'value'})
+    else:
+        expected = pd.DataFrame(
+            {'id': list(range(0, 10)) * 2, 'str_col': 'value'}
+        )
+
+    result = result.sort_values(['id']).reset_index(drop=True)
```

Comment:
Do we need to sort here? I expect Spark to return

[0, 1, 2, 9, 0, 1, 2,... 9] Which should match "expected"?

Reply:
When `distinct=True` the rows will be out of order (`distinct=True` -> compilation does PySpark `.distinct()` which seems to not retain order)